### PR TITLE
feat: detect *.jar files as maven

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
- allow detecting `*.jar` files to be processed by the  mvn plugin 
- split out test, monitor and protect tests into separate files

#### What are the relevant tickets?
[BST-1044](https://snyksec.atlassian.net/browse/BST-1044)

### CODE REVIEW

File: test/acceptance/protect.test.ts [hunk 1]
```diff
@@ -26,3 +80,92 @@ test('`protect` should not fail for unauthorized users', (t) => {
     userConfig.set('api', apiUserConfig);
   });
 });
+
+test('`protect` for unsupported package managers', async (t) => {
```
Comment (lili2311) [inline test/acceptance/protect.test.ts]: copied these over from main acceptance test

File: src/lib/detect.ts [hunk 2]
```diff
@@ -40,6 +40,7 @@ const DETECTABLE_PACKAGE_MANAGERS: {
   '.gemspec': 'rubygems',
   'package-lock.json': 'npm',
   'pom.xml': 'maven',
+  '.jar': 'maven',
```
Comment (orsagie) [inline src/lib/detect.ts]: I think we also want to detect .war

File: test/acceptance/cli-monitor.acceptance.test.ts [hunk 3]
```diff
@@ -0,0 +1,1584 @@
+import * as tap from 'tap';
```
Comment (orsagie) [inline test/acceptance/cli-monitor.acceptance.test.ts]: This makes me so happy 😄

File: src/lib/detect.ts (hunk 2 above)
Comment (orsagie) [reply src/lib/detect.ts]: and `zip`, and perhaps `ear`?

Comment (lili2311) [reply src/lib/detect.ts]: let's add the rest as the customer asks for this, added `jar` and `war`

**Pull request merged**